js/tsconfig.json

```diff
@@ -0,0 +1,13 @@
+{
+    "compilerOptions": {
+        "esModuleInterop": true,
+        "exactOptionalPropertyTypes": false,
+        "experimentalDecorators": true,
+        "forceConsistentCasingInFileNames": true,
+        "noUncheckedIndexedAccess": false,
+        "incremental": true
```

Comment:
Genuine question (I am not a TypeScript expert): do we need all of these if we already extend node18 and strictest config?

Reply:
Probably not. I carried them from another project.
I've removed them all and fixed a bit of test code that assumed "noUncheckedIndexAccess": false.